source_handlers_test.go

```diff
@@ -1136,3 +1139,113 @@ func sourceEventTestHelper(t *testing.T, c echo.Context, expectedEventType strin
 
 	return nil
 }
+
+// TestSourceEditPausedIntegration tests that a "bad request" response is returned when a paused source is tried to
+// be updated. Runs on integration tests.
+func TestSourceEditPausedIntegration(t *testing.T) {
+	testutils.SkipIfNotRunningIntegrationTests(t)
+
+	newSourceName := "New source name"
+	req := m.SourceEditRequest{
+		Name:               request.PointerToString(newSourceName),
+		AvailabilityStatus: request.PointerToString("available"),
+	}
+
+	body, _ := json.Marshal(req)
+
+	c, rec := request.CreateTestContext(
+		http.MethodPatch,
+		"/api/sources/v3.1/sources/1",
+		bytes.NewReader(body),
+		map[string]interface{}{
+			"tenantID": int64(1),
+		},
+	)
+
+	c.SetParamNames("id")
+	c.SetParamValues("1")
+	c.Request().Header.Add("Content-Type", "application/json;charset=utf-8")
+
+	// Modify the source so that the underlying code identifies it as "paused".
+	err := dao.DB.Model(m.Source{}).Where("id = ?", 1).UpdateColumn("paused_at", time.Now()).Error
+	if err != nil {
+		t.Error(err)
+	}
+
+	badRequestSourceEdit := ErrorHandlingContext(SourceEdit)
+	err = badRequestSourceEdit(c)
+	if err != nil {
+		t.Error(err)
+	}
+
+	// Revert the changes so other tests don't have any problems.
+	err = dao.DB.Model(m.Source{}).Where("id = ?", 1).UpdateColumn("paused_at", nil).Error
+	if err != nil {
+		t.Error(err)
+	}
+
+	if rec.Code != http.StatusBadRequest {
+		t.Errorf("Wrong return code, expected %v got %v", http.StatusBadRequest, rec.Code)
+	}
+
+	want := "can't modify a paused source. Unpause it first and try again"
+	got := rec.Body.String()
+	if !strings.Contains(got, want) {
+		t.Errorf(`unexpected body returned. Want "%s" contained in what we got "%s"`, want, got)
+	}
+}
+
+// TestSourceEditPausedUnit tests that a "bad request" response is returned when a paused source is tried to
+// be updated. Runs on unit tests.
+func TestSourceEditPausedUnit(t *testing.T) {
+	testutils.SkipIfNotRunningIntegrationTests(t)
```

Comment:
In comment is "Runs on unit tests" and then you skip if unit tests are running

Reply:
You are right, I copied and pasted the test, and forgot to modify it :facepalm: now the unit tests runs on unit tests :laughing: